testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/handlers/RequestDumpingHandlerTestCase.java

```diff
@@ -0,0 +1,299 @@
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        // Remove custom server log setting
+        ManagementOperations.executeOperation(mgmtClient.getControllerClient(),
+                Util.getWriteAttributeOperation(aLogAddr, "file", originalValue));
+
+        // Delete custom server log file
+        Files.delete(logFilePath);
+    }
+
+    private final static String DEPLOYMENT = "no-req-dump";
+    private final static String DEPLOYMENT_DUMP = "req-dump";
+
+    @Deployment(name = DEPLOYMENT_DUMP)
+    public static WebArchive deployWithReqDump() {
+        WebArchive war = ShrinkWrap
+                .create(WebArchive.class, DEPLOYMENT_DUMP + ".war")
+                .addPackage(RequestDumpingHandlerTestCase.class.getPackage())
+                .addAsWebInfResource(RequestDumpingHandlerTestCase.class.getPackage(), "jboss-web-req-dump.xml",
+                        "jboss-web.xml").addAsWebResource(new StringAsset("A file"), "file.txt");
+        return war;
+    }
+
+    @Deployment(name = DEPLOYMENT)
+    public static WebArchive deployWithoutReqDump() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, DEPLOYMENT + ".war")
+                .addPackage(RequestDumpingHandlerTestCase.class.getPackage())
+                .addAsWebInfResource(RequestDumpingHandlerTestCase.class.getPackage(), "jboss-web.xml", "jboss-web.xml")
+                .addAsWebResource(new StringAsset("A file"), "file.txt");
+        return war;
+    }
+
+    /**
+     * Testing app has already defined request dumper handler. This test checks that when an request to URL is performed then
+     * request detail data is stored in proper format in the proper log file. Also check that warning for the user is generated.
+     */
+    @Test
+    @OperateOnDeployment(DEPLOYMENT_DUMP)
+    public void testReqDumpHandlerOn(@ArquillianResource URL url) throws Exception {
+        commonTestBody(url, true);
+    }
+
+    /**
+     * Testing app has no request dumper handler registered. This test checks that there is not dump additional info about
+     * executed request in the logfile. Also no warning is shown to user.
+     */
+    @Test
+    @OperateOnDeployment(DEPLOYMENT)
+    public void testReqDumpHandlerOff(@ArquillianResource URL url) throws Exception {
+        commonTestBody(url, false);
+    }
+
+    /**
+     * Common test body part.
+     *
+     * @param url deployment url
+     * @param requestDumperOn whether RequestDumpingHandler is turned on
+     * @throws Exception
+     */
+    private void commonTestBody(@ArquillianResource URL url, boolean requestDumperOn) throws Exception {
+        // Set custom system log name
+        ModelNode file = new ModelNode();
+        file.get("relative-to").set(relativeTo);
+        file.get("path").set(logFilePrefix + LOG_FILE_SUFFIX);
+        ModelNode op = Util.getWriteAttributeOperation(aLogAddr, "file", file);
```

Comment:
This feels very volatile, as it changes log location for whole server. 
I think it would be better if test would add new logger (io.undertow.request.dump) and append only to new file handler that would than have only request logs in it.
Also the logic for looking if content is in file would be much easier.